src/ProjectTemplates/Web.ProjectTemplates/content/EmptyBlazorServerWeb-CSharp/Program.cs

```diff
@@ -0,0 +1,30 @@
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.Web;
+
+var builder = WebApplication.CreateBuilder(args);
+builder.Services.AddRazorPages();
+builder.Services.AddServerSideBlazor();
+
+var app = builder.Build();
+
+if (!app.Environment.IsDevelopment())
+{
+#if (RequiresHttps)
+    // The default HSTS value is 30 days. You may want to change this for production scenarios, see https://aka.ms/aspnetcore-hsts.
+    app.UseHsts();
+}
+
+app.UseHttpsRedirection();
+#else
+}
+
+#endif
```

Comment:
I believe this may actually lead to different behavior than our current blazorserver template:

https://github.com/dotnet/aspnetcore/blob/0ddda008079da4bdb81ff9bdbd9621dec89e0a53/src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Program.cs#L123-L144


```suggestion
#if (RequiresHttps)
if (!app.Environment.IsDevelopment())
{
    // The default HSTS value is 30 days. You may want to change this for production scenarios, see https://aka.ms/aspnetcore-hsts.
    app.UseHsts();
}

app.UseHttpsRedirection();
#endif
```